state-sync/inter-component/storage-service-notifications/src/lib.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use aptos_channels::{self, aptos_channel, message_queues::QueueStyle};
+use async_trait::async_trait;
+use futures::{stream::FusedStream, Stream};
+use serde::{Deserialize, Serialize};
+use std::{
+    fmt,
+    pin::Pin,
+    task::{Context, Poll},
+};
+use thiserror::Error;
+
+// TODO: do we want to make this configurable?
+const STORAGE_SERVICE_NOTIFICATION_CHANNEL_SIZE: usize = 1;
+
+#[derive(Clone, Debug, Deserialize, Error, PartialEq, Eq, Serialize)]
+pub enum Error {
+    #[error("Commit notification failed: {0}")]
+    CommitNotificationError(String),
+}
+
+/// The interface between state sync and the storage service, allowing state sync to notify
+/// the storage service of events (e.g., newly committed transactions).
+#[async_trait]
+pub trait StorageServiceNotificationSender: Send + Clone + Sync + 'static {
+    /// Notify the storage service of newly committed transactions
+    /// at the specified version.
+    async fn notify_new_commit(&self, highest_synced_version: u64) -> Result<(), Error>;
+}
+
+/// This method returns a (StorageServiceNotifier, StorageServiceNotificationListener) pair
+/// that can be used to allow state sync and the storage service to communicate.
+///
+/// Note: state sync should take the notifier and the storage service should take the listener.
+pub fn new_storage_service_notifier_listener_pair(
+) -> (StorageServiceNotifier, StorageServiceNotificationListener) {
+    // Create a dedicated channel for notifications
+    let (notification_sender, notification_receiver) = aptos_channel::new(
+        QueueStyle::FIFO,
```

Comment:
Shouldn't this technically be LIFO? Maybe it doesn't matter because we don't actually use the value, but maybe the trace log will be confusing if we see old versions?

Reply:
Yes, good call. 🚀  Technically, the value isn't used, but I've updated it to be `LIFO` (on both queues) so that the trace logs make sense 😄